### ISSUE TITLE
[bug] Phalcon\Paginator\Adapter\QueryBuilder count is not accurate

### ISSUE THREAD

**Issue by zhouyl** [phalcon/cphalcon#2411]
(Sorry, i am chinese! Too bad English.)

When QueryBuider presence group by, count is not accurate.

My Code

```php
class MyQueryBuilder extends \Phalcon\Paginator\Adapter\QueryBuilder
{

    public function getPaginate()
    {
        $paginate = parent::getPaginate();

        $builder = $this->getQueryBuilder();

        if ($group = $builder->getGroupBy()) {
            $result = $builder->columns('count(distinct ' . $group . ') as total')
                ->groupBy(null)
                ->getQuery()
                ->execute()
                ->getFirst();
            $paginate->total_items = $result->total;
            $paginate->total_pages = ceil($paginate->total_items / $this->getLimit());
        }

        return $paginate;
    }

}
```

**Comment by braf** [comment]
Very sorry about referencing this issue a billion times, I was committing from a vagrant box and the author kept coming out wrong and I was fixing it with some rebasing and didn't realise it would do this from my own fork. :cry:

**Comment by andresgutierrez** [comment]
@braf :+1: no problem

**Comment by Green-Cat** [comment]
@sergeyklay Fixed in #11195
